apps/prairielearn/src/pages/externalImageCapture/externalImageCapture.ts

```diff
@@ -0,0 +1,76 @@
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+
+import { HttpStatusError } from '@prairielearn/error';
+
+import { emitExternalImageCapture } from '../../lib/externalImageCaptureSocket.js';
+import { selectAndAuthzVariant } from '../../models/variant.js';
+
+import { ExternalImageCapture } from './externalImageCapture.html.js';
+
+const router = express.Router({ mergeParams: true });
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!req.query.file_name) {
+      throw new HttpStatusError(400, 'file_name query parameter is required');
+    }
+    res.send(
+      ExternalImageCapture({
+        variantId: req.params.variant_id,
+        fileName: req.query.file_name as string,
+        resLocals: res.locals,
+      }),
+    );
+  }),
+);
+
+// Handles image uploading from the external image capture page.
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const variantId = req.params.variant_id as string;
+    const fileName = req.query.file_name as string;
+
+    const variant = await selectAndAuthzVariant({
+      unsafe_variant_id: variantId,
+      variant_course: res.locals.course,
+      question_id: res.locals.question.id,
+      course_instance_id: res.locals?.course_instance?.id,
+      instance_question_id: res.locals.instance_question?.id,
+      authz_data: res.locals.authz_data,
+      authn_user: res.locals.authn_user,
+      user: res.locals.user,
+      is_administrator: res.locals.is_administrator,
+      publicQuestionPreview: res.locals.public_question_preview,
+    });
+
+    if (!variant) {
+      throw new HttpStatusError(404, 'Variant not found');
+    }
+
+    if (!variant.open) {
+      throw new HttpStatusError(403, 'This variant is not open');
+    }
```

Comment:
This validation should probably be done in the GET as well, or in a `router.use` before it.